AD9081: Frequency Hopping Support

mhennerich opened the pull request.

nunojsa commented on drivers/iio/adc/ad9081.c:
```diff
@@ -968,6 +978,84 @@ static const struct iio_enum ad9081_nyquist_zone_enum = {
+
+	if (chan->output)
+		return phy->tx_ffh_hopf_mode[cddc_num];
+	else
```
redundant else...

nunojsa commented on drivers/iio/adc/ad9081.c:
```diff
@@ -1100,14 +1188,64 @@ static ssize_t ad9081_ext_info_read(struct iio_dev *indio_dev,
+			break;
+		}
+		ret = sprintf(buf, "%lld\n",
+			phy->tx_ffh_hopf_vals[phy->tx_ffh_hopf_index[cddc_num]][cddc_num]);
```
`sysfs_emit()`

nunojsa reviewed (commented):
Also, it seems there are valid checkpatch stuff

mhennerich commented:
Pushed V2:

- Use sysfs_emit
- Remove return patch
- Added separate FFH devicetree
- Added patch to support AD9177, AD9207 & AD9209 chip variants 
- Cleaned some checkpatch issues
